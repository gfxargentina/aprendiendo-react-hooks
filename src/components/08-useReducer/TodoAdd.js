import React from 'react'
import { useForm } from '../../hooks/useForm';


const TodoAdd = ({ handleAddTodo }) => {

    const [ { descripcion }, handleInputChange, reset ] =  useForm({
        descripcion: ''
    });

    const handleSubmit = (e) => {
        e.preventDefault();
          //validacion form
          if ( descripcion.trim().length <= 1 ) {
            return;
          }
  
         //console.log('nueva tarea')
         const newTodo = {
          id: new Date().getTime(),
          descripcion: descripcion,
          done: false
         };
  
         handleAddTodo( newTodo );
         reset();
      }

    return (
        <>
            <h4>Agregar Todo</h4>
                        <hr/>

                        <form onSubmit={ handleSubmit }  >
                            <input 
                            type="text"
                            className="mb-3"
                            name="descripcion"
                            placeholder="descripcion"
                            value={ descripcion }
                            onChange={ handleInputChange }
                            />

                            <button 
                                type="submit"
                                className="btn btn-outline-primary btn-block" >
                                Agregar
                            </button>

                        </form>
            
        </>
    )
}

export default TodoAdd
