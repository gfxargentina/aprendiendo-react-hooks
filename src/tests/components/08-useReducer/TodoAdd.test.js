import React from 'react';
import  {shallow}  from 'enzyme';
import  TodoAdd  from '../../../components/08-useReducer/TodoAdd';
//import  {demoTodos}  from '../../fixtures/demoTodos';

describe('Pruebas en <TodoAdd />', () => {
    const handleAddTodo = jest.fn();

    const wrapper = shallow(
        <TodoAdd 
            handleAddTodo= { handleAddTodo }
        />
    )

    test('debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
        
    })
    test('NO debe de llamar handleAddTodo', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });
        expect( handleAddTodo ).toHaveBeenCalledTimes(0);
        
    })

    test('debe de llamar la funcion handleAddTodo', () => {

        const value = 'Aprendiendo Testing 2';
        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'descripcion'
            }
        });
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });

        expect( handleAddTodo ).toHaveBeenCalledTimes(1);
        expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object) );
        //espera que envia cualquier numero para el id
        expect( handleAddTodo ).toHaveBeenCalledWith({
            id: expect.any(Number),
            descripcion: value,
            done: false
        }
            );
        //boton reset se espera que limpie los campos y no haiga nada
            expect( wrapper.find('input').prop('value') ).toBe('');

    })
    
    
    
    
})

