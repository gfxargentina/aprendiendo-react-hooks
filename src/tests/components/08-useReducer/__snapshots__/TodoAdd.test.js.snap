// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      className="mb-3"
      name="descripcion"
      onChange={[Function]}
      placeholder="descripcion"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
